src/System.Private.CoreLib/src/System/Threading/ManagedThreadId.cs

```diff
@@ -36,88 +57,77 @@ public static int Current
             }
         }
 
-        private static volatile ManagedThreadId s_list;
-        private static int s_nextThreadId;
+        private static int MakeForCurrentThread()
+        {
+            int newThreadId = Environment.ManagedThreadIdNone;
 
-        private readonly int _managedThreadId; // The ID represented by this object.
-        private volatile int _lock;            // 1 if a thread is trying to recycle this ID; 0 otherwise.
-        private IntPtr _nativeThreadHandle;    // A handle to the thread that currently owns this ID; the thread may be dead.
-        private ManagedThreadId _next;
+            for (RecycledIdsList node = s_recycledIdsList; node != null; node = node._next)
+            {
+                if (Interlocked.Increment(ref node._lock) != 0)
+                    continue;
 
-        public ManagedThreadId(int id)
-        {
-            _managedThreadId = id;
+                try
+                {
+                    if (node._id == Environment.ManagedThreadIdNone)
```

Comment:
Could you please move the definition of ManagedthreadIdNone to this file while you are on it? It is odd for it to be in Environment class.